app/jobs/remove_old_throttles_job.rb

```diff
@@ -20,11 +20,13 @@ def perform(now, limit: 1000, total_limit: 100_000)
     total_removed = 0
 
     loop do
-      removed_count = Throttle.
-        where('updated_at < ?', now - (WINDOW + max_window.minutes)).
-        or(Throttle.where(updated_at: nil)).
-        limit(limit).
-        delete_all
+      removed_count = write_transaction_with_timeout do
+        Throttle.
+          where('updated_at < ?', now - (WINDOW + max_window.minutes)).
+          or(Throttle.where(updated_at: nil)).
+          limit(limit).
```

Comment:
another option would be to decrease the limit, right?

Reply:
that's a much better idea, I will drop all of these changes and halve (?) both limits